File: sample/src/main/java/com/chuckerteam/chucker/sample/SampleApplication.java
package com.chuckerteam.chucker.sample;

import android.app.Application;

import com.squareup.leakcanary.LeakCanary;

public class SampleApplication extends Application {
    @Override public void onCreate() {
        super.onCreate();
        initLeakCanary();
    }

    private void initLeakCanary() {
        if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return;
        }
        LeakCanary.install(this);
    }
}

File: sample/src/main/kotlin/com/chuckerteam/chucker/sample/SampleApplication.kt
package com.chuckerteam.chucker.sample

import android.app.Application
import com.chuckerteam.chucker.api.ChuckerCollector
import com.chuckerteam.chucker.api.RetentionManager
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import kotlinx.coroutines.launch

class SampleApplication : Application() {

    private val serverScope = CoroutineScope(Dispatchers.IO + SupervisorJob())
    private val grpcServer by lazy { SampleGrpcServer(GRPC_PORT) }

    val chuckerCollector by lazy {
        ChuckerCollector(
            context = this,
            showNotification = true,
            retentionPeriod = RetentionManager.Period.ONE_HOUR,
        )
    }

    override fun onCreate() {
        super.onCreate()
        serverScope.launch { grpcServer.start() }
    }

    override fun onTerminate() {
        super.onTerminate()
        serverScope.launch { grpcServer.stop() }
    }

    companion object {
        const val GRPC_PORT = 50051
    }
}
